lib/utils/report.cjs

```diff
@@ -91,14 +93,24 @@ function report(problem) {
 
 	if (start) {
 		warningProperties.start = start;
-	} else if (validateTypes.isNumber(index)) {
-		warningProperties.index = index;
+	} else if (range?.start) {
+		warningProperties.start = range.start;
+	} else if (validateTypes.isNumber(startLine)) {
+		warningProperties.start = {
+			line: startLine,
+			column: 1, // use the start of the line when `node` is undefined
+		};
 	}
 
 	if (end) {
 		warningProperties.end = end;
-	} else if (validateTypes.isNumber(endIndex)) {
-		warningProperties.endIndex = endIndex;
+	} else if (range?.end) {
```

Comment:
question
nit

Since we are sure that if `range.end` is present it's always accompanied by `range.start`, should we instead add another nesting?
i.e.
```js
if (range) {
  warningProperties.start = range.start;
  warningProperties.end = range.end;
} else { …
```
Does it even matter?
I mean, do we want to allow a form of overriding by passing either `start` and/or `end` to `report`?
i.e. is this by design?

Reply:
I am unsure.
I find the current shape of the code very hard to reason about :)

It is unclear to me which exact combinations of optional values we support and what the expected outcome is for each.

I've filed https://github.com/stylelint/stylelint/pull/8098 with a possible improvement, although that still doesn't fully address the ambiguity.

Reply:
wait a second

It was just a nit, damnit :)
Ill take a look to the other PR.

Reply:
I filed that PR before you nitted 😛 
It was something I was already thinking about.